### PR TITLE
(PUP-11768) Layout cops

### PR DESCRIPTION
**Pull request by mhashizume**
This PR comprises another series of commits to enable previously-disabled Rubocop Layout cops.

### CODE REVIEW

File: lib/puppet/pops/model/ast.rb [hunk 1]
```diff
@@ -4755,7 +4755,8 @@ def self.register_pcore_types
     Model::SelectorEntry,
     Model::SelectorExpression,
     Model::NamedAccessExpression,
-    Model::Program]
+    Model::Program
+  ]
```
Comment (joshcooper) [inline lib/puppet/pops/model/ast.rb:4810]: This file is autogenerated via the generate_ast_model rake task https://github.com/puppetlabs/puppet/blob/272abb97efa5d1c44fd238e3988b5a9f7e87cd3f/tasks/generate_ast_model.rake#L85 Maybe just ignore `lib/puppet/pops/model/ast.rb`?

File: lib/puppet/pops/model/ast.rb [hunk 2]
```diff
@@ -89,7 +89,8 @@ def self.from_asserted_hash(init_hash)
     new(
       init_hash['locator'],
       init_hash['offset'],
-      init_hash['length'])
+      init_hash['length']
+    )
```
Comment (joshcooper) [inline lib/puppet/pops/model/ast.rb:93]: Same comment about ignoring this file

File: .rubocop.yml [hunk 3]
```diff
@@ -39,6 +39,9 @@ Layout/IndentationWidth:
   Exclude:
     - 'lib/puppet/provider/package/*.rb'
 
+Layout/LineEndStringConcatenationIndentation:
+  Enabled: true
+
```
Comment (joshcooper) [inline .rubocop.yml:31]: I think the default is to enable the cop, so maybe it doesn't need to be explicitly enabled here

Comment (mhashizume) [reply .rubocop.yml:31]: I believe that for new cops, they're set as pending until they're explicitly enabled: https://docs.rubocop.org/rubocop/versioning.html#pending-cops

When I run Rubocop locally, I get this:

```
$ bundle exec rubocop
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

Please also note that you can opt-in to new cops by default by adding this to your config:
  AllCops:
    NewCops: enable
```

Followed by a list of pending cops

File: lib/puppet/pops/model/ast.rb (hunk 1 above)
Comment (mhashizume) [reply lib/puppet/pops/model/ast.rb]: Thanks for the review. I've added a new commit to exclude this file from Rubocop checks and have rolled back the changes I'd made to the file as well.

File: .rubocop.yml (hunk 3 above)
Comment (joshcooper) [reply .rubocop.yml:32]: Ahh ok that makes sense

**Review by joshcooper — APPROVED** [review]

**Pull request merged**